### PR TITLE
GH-270: Fix failing tests in the PR of the same ID

### PR DESCRIPTION
**Pull request by qtran-n4**
N4*AvoidRefreshDocumentProviderPluginUITest were failing. In this fix, we only observe POST CHANGE event on the main thread to avoid race condition caused by multi-threads

### CODE REVIEW

**Comment by qtran-n4** [comment]
@yoosiba Please review the fix again. Thanks! Waiting for the build...

File: testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/tests/helper/documentprovider/CountPostChangeBroadcastChangeNotificationManager.xtend [hunk 1]
```diff
@@ -20,16 +20,20 @@ import org.eclipse.core.resources.IResourceChangeEvent
...
 		super(workspace)
+		this.mainThread = mainThread
 	}
 
 	override void broadcastChanges(ElementTree lastState, ResourceChangeEvent event, boolean lockTree) {
-		if (event.type == IResourceChangeEvent.POST_CHANGE)
+		// Count the post change events triggered on the main thread
+		if (Thread.currentThread == mainThread && event.type == IResourceChangeEvent.POST_CHANGE) {
 			countPostChangeBroadcastTriggered++
+			println(Thread.currentThread.stackTrace)
```
Comment (yoosiba) [inline testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/tests/helper/documentprovider/CountPostChangeBroadcastChangeNotificationManager.xtend]: Use logger and `WARN` or `DEBUG` level. Do not pollute the logs if no need.

Comment (qtran-n4) [reply testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/tests/helper/documentprovider/CountPostChangeBroadcastChangeNotificationManager.xtend]: OOps it was for debugging I forget to remove it

**Comment by qtran-n4** [comment]
Thanks! The build is green so I am merging.

**Pull request merged**